common/types/activation.go

```diff
@@ -135,10 +142,40 @@ func (atxh *ActivationTxHeader) SetID(id *ATXID) {
 	atxh.id = id
 }
 
-// GetWeight returns the ATX's weight = numUnits * ticks.
+// GetWeight of the atx.
+// Will panic if read before Verify is called.
 func (atxh *ActivationTxHeader) GetWeight() uint64 {
-	// TODO: Limit the number of bits this can occupy
-	return uint64(atxh.NumUnits) * (atxh.EndTick - atxh.StartTick)
+	return uint64(atxh.NumUnits) * (atxh.tickCount)
+}
+
+// BaseTickHeight is a tick height of the positional atx.
+// Will panic if read before Verify is called.
+func (atxh *ActivationTxHeader) BaseTickHeight() uint64 {
```

Comment:
i don't understand why this BaseTickHeight() and TickCount() is needed. once we verify an ATX's tick height, why can't we just record the sum in the db instead of recording them separately?

Reply:
see reply below. it is not possible to merge them, we are using both of them for different use cases